src/main/java/uk/co/ramp/covid/simulation/lockdown/LocalLockdownEventGenerator.java

```diff
@@ -0,0 +1,92 @@
+package uk.co.ramp.covid.simulation.lockdown;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import uk.co.ramp.covid.simulation.population.Person;
+import uk.co.ramp.covid.simulation.util.Time;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class LocalLockdownEventGenerator extends LockdownEventGenerator {
+    private static final Logger LOGGER = LogManager.getLogger(LocalLockdownEventGenerator.class);
+
+    private Double newCasesThreshold = null;
+    private Integer numHospitalisedThreshold = null;
+
+    private Double socialDistance = null;
+    
+    private int prevInfected = 0;
+    private boolean gotInitialInfections = false;
+
+    private int getNumInfected() {
+        int inf = 0;
+        for (Person p : population.getAllPeople()) {
+            if (p.isinfected()) {
+                inf++;
+            }
+        }
+        return inf;
+    }
+
+    private int getNumHospitalised() {
+        int inHospital = 0;
+        for (Person p : population.getAllPeople()) {
+            if (p.isHospitalised()) {
+                inHospital++;
+            }
+        }
+        return inHospital;
+    }
+
+    private List<LockdownEvent> handleNewCasesLockdown(Time t) {
+        List<LockdownEvent> newEvents = new ArrayList<>();
+
+        int infected = getNumInfected();
+        int newCases = infected - prevInfected;
+        prevInfected = infected;
+
+        // We need to skipt the first day check so we can get the *new* cases
```

Comment:
Is the first day not always Time(0)?

Reply:
This should be "first day the generator is active" since we might not start looking for these conditions till day 200 (for example). I've improved this comment a little.